common/app/routes/Challenges/redux/modal-epic.js

```diff
@@ -0,0 +1,132 @@
+import { Observable } from 'rx';
+import { ofType } from 'redux-epic';
+import { types, chatRoomSelector } from '../redux';
+
+import { filesSelector } from '../../../files';
+import { currentChallengeSelector } from '../../../redux';
+
+function filesToMarkdown(files = {}) {
+  const moreThenOneFile = Object.keys(files).length > 1;
+  return Object.keys(files).reduce((fileString, key) => {
+    const file = files[key];
+    if (!file) {
+      return fileString;
+    }
+    const fileName = moreThenOneFile ? `\\ file: ${file.contents}` : '';
+    const fileType = file.ext;
+    return fileString +
+      '\`\`\`' +
+      fileType +
+      '\n' +
+      fileName +
+      '\n' +
+      file.contents +
+      '\n' +
+      '\`\`\`\n\n';
+  }, '\n');
+}
+
+export default function bugEpic(actions, { getState }, { window }) {
+  return actions::ofType(
+    types.openIssueSearch,
+    types.createIssue,
+    types.createQuestion,
+    types.openHelpChatRoom
+  )
+    .map(({ type }) => {
+      const state = getState();
+      const files = filesSelector(state);
+      const challengeName = currentChallengeSelector(state);
+      const helpChatRoom = chatRoomSelector(state);
+      const {
+        navigator: { userAgent },
+        location: { href }
+      } = window;
+
+      switch (type) {
```

Comment:
This imperative switch statement is unnecessary. This can be done declaratively using separate epics that respond to a single action type. This makes testing epics much easier and makes the logic easier to follow.

Reply:
Noted and fixed.